tests/test_csr.py

```diff
@@ -451,3 +449,44 @@ def test_saving_public_certifate_disables_signing_request_fields(admin_client):
     # csr, public_certificate, common_name, organization_name, country_name,
     # state_or_province_name, locality_name, email_address, certificate
     assert response.content.decode().count("readonly") == 9
+
+
+@pytest.mark.django_db
+def test_csr_display(admin_client, client):
+    add_url = reverse("admin:simple_certmanager_signingrequest_add")
+
+    response = admin_client.get(add_url)
+
+    assert response.status_code == 200
+    assert "CSR not generated yet" in response.content.decode()
+
+    data = {
+        "common_name": "test.com",
+        "country_name": "US",
+        "organization_name": "Test Org",
+        "state_or_province_name": "Test State",
+        "email_address": "test@test.com",
+    }
+
+    response = admin_client.post(add_url, data, follow=True)
+
+    assert response.status_code == 200
+
+    signing_request = SigningRequest.objects.get()
+    assert signing_request is not None
+    assert signing_request.private_key != ""
+    assert signing_request.csr != ""
+    assert "BEGIN PRIVATE KEY" in signing_request.private_key
+    assert "Download CSR" in response.content.decode()
+
+    # Download CSR logged in as admin
+    download_csr_url = reverse("admin:download_csr", args=(signing_request.pk,))
+    response = admin_client.get(download_csr_url)
+    assert response.status_code == 200
+
+    # User without permission to add SigningRequests can't download CSR
+    response = client.get(download_csr_url)
+    assert response.status_code == 302
+    base_url = "/admin/login/?next=/admin/simple_certmanager/"
+    url = base_url + "signingrequest/dowload_csr/{}".format(signing_request.pk)
+    assert response.url == url
```

Comment:
I feel this test asserts a lot of unrelated. Can it be logically be split up in smaller tests? Or am I misreading and in this sequence a unit?